Fix label load old jasp file

JorisGoosen opened the pull request:
Several tweaks to JASP data loading
- Missing values should be hierarchical so workspace missing values should be used in addition to the ones from a column. See the stata test file, it doesnt contain specific `nan` missing value specifications for the columns, but it really ought to be a missing value as it is so defined anyway in the workspace. This change also allows one to disable and reenable custom missing values without losing them.
- Make sure labels imported from an old (<0.18) jasp file are sorted on load
- added a testcase for loading 0.16.4 and 0.18.3 jasp file (0.17.3 app didnt work... because apple said no)
- Small change to how titles are set to make loading an old jaspfile more like making a new one.
Fixes https://github.com/jasp-stats/jasp-issues/issues/3710
- a test has been added for verifying that titles adhere to namechanges correctly
- a test has been added for verifying that empty values are adhered to, both as workspace and non-workspace

Loading debug jasp file from 0.18.3 has some inconsistencies now with loading both older and newer files where:
- autoSort is disabled
- thus the labels are sorted differently
- the title is set to name regardless of the change made for the older files.

JorisGoosen committed:
No need to do a dbload when loading an old jasp file
Also make sure to update nonempty indexes etc

Fixes https://github.com/jasp-stats/jasp-issues/issues/3710

JorisGoosen committed:
add test for checking data import from jaspfiles

also write nonEmptyLabels because those were wrong for the issue solved in the last commit

JorisGoosen committed:
handle emptyvalues more logically and dont drop data

JorisGoosen committed:
add 0.16.4 debug jasp file and add updated json for stata sample

JorisGoosen committed:
labels should be sorted when loaded from old jasp file
also make sure that _title does not contain unneeded double bookkeeping
(if _name == _title _title really ought to be "" and not _name)

JorisGoosen committed:
update json for debug 0.16.4

JorisGoosen committed:
make sure the new retitle code keeps working by adding a unittest to debugdatasettester

JorisGoosen committed:
add testcase for emptyvalues

JorisGoosen committed:
make sure the test also combines for a combination of workspace and column empty values

boutinb commented:
If I load the hypoTherapic jasp file from https://github.com/jasp-stats/jasp-issues/issues/3710, I get all labels, but if I refresh the analyses, I get an error: 
<img width="1304" height="482" alt="Screenshot 2025-10-29 at 12 57 27" src="https://github.com/user-attachments/assets/c854c7e0-c419-4f80-b6c3-dd4fc8de7e02" />

JorisGoosen commented:
Ill have a look!

JorisGoosen commented:
(I guess the numeric things arent reset properly either? Sounds like a good testcase to add in any case)

JorisGoosen committed:
store all labels after loading old jasp file
also check for each data import whether what it writes to sqlite is the same as it contains by loading and comparing

JorisGoosen committed:
So the -1 was messing things up

Not sure if there are still files around that do use a 1-based int storage?

JorisGoosen committed:
also add the 0.17.3 created file as a testcase as it contains nothing but 1 and 0s anyway

JorisGoosen commented:
So labels werent being stored, and then when they werent they were getting bogus intsId (starting at -1)

Now it should work
